hera_pspec/pspecdata.py

```diff
@@ -727,42 +760,78 @@ def scalar(self, stokes='I', taper='none', little_h=True, num_steps=2000):
-              taper='none', little_h=True, verbose=True):
+    def pspec(self, bls1, bls2, dset1_ind, dset2_ind, input_data_weight='identity', norm='I', taper='none', 
+              little_h=True, avg_groups=False, exclude_auto_bls=False, exclude_conjugated_blpairs=False,
+              verbose=True):
         """
         Estimate the delay power spectrum from the datasets contained in this 
         object, using the optimal quadratic estimator from arXiv:1502.06016.
 
+        In this formulation, the power spectrum is proportional to the visibility data via
+
+        P = M data_{LH} E data_{RH}
+
+        where E contains the data weighting and FT matrices, M is a normalization matrix, 
+        and the two separate datasets are denoted as "left-hand" and "right-hand".
+
+        Each power spectrum is generated by taking a bl from bls1 out of the the dset1_ind'th dataset
+        and assigning it as data_LH, and a bl from bls2 out of the dset2_ind'th dataset and assigning it
+        as data_RH.
+
         Parameters
         ----------
-        bls : list of tuples (or lists of tuples)
-            List of baselines to include in the power spectrum calculation. 
-            Each baseline is specified as a tuple of antenna IDs.
+        bls1 : list of bl tuples (or list of bl groups, which themselves are list of bl tuples)
+            List of baseline tuples to use in the power spectrum calculation for the 
+            "left-hand" dataset. A baseline tuple is specified as an antenna pair, Ex: (ant1, ant2)
             
-            If an element of the list is a list of tuples, the baselines in 
-            that list will be averaged together in the power spectrum 
-            calculation, reducing the number of cross-correlations that are 
-            needed.
+        bls2 : list of bl tuples (or list of bl groups, which themselves are list of bl tuples)
+            List of baseline tuples to use in the power spectrum calculation for the 
+            "right-hand" dataset. A baseline tuple is specified as an antenna pair, Ex: (ant1, ant2)
+
+            Alternatively, bls1 and bls2 can contain lists of baselines groups, which are themselves lists of
+            baseline tuples. A bl-group in bls1 should have the same positional index in bls2.
+            In this case, pspec will do one of two things:
+
+                1) The data in each baseline-group are averaged together before squaring (avg_groups=True),
+                   and then crossed with the same positional baseline group between bls1 and bls2.
+
+                2) All permutations of cross-spectra between bls in a group in bls1 and bls in a group
+                   in bls2 are calculated (avg_groups=False). Note that baselines are never crossed between
+                   different baseline groups, which are defined based on their positional index.
+
+                   If exclude_auto_bls == True, bl-pairs that have a repeated baseline are eliminted.
```

Comment:
Typo.
Also, can this be expanded a bit to include an example, and to define what happens with groups? What happens when `avg_groups=True` and the two groups contain the same baseline?

Reply:
yup! i can add an example for `exclude_auto_bls`. 

ah, average groups doesn't average between groups, it just means to average within a group.. perhaps that should be made more clear..

Reply:
i'm going to change the keywarg to `avg_group`, which i think is a little truer to what it's doing (even though there may be multiple groups, each of which get averaged independently)